fontstash/fontstash.h

```diff
@@ -479,10 +490,10 @@ struct FONSshapingRes
 typedef struct FONSshapingRes FONSshapingRes;
 
 struct FONSshaping {
-    FONSshapingRes* shapingRes;
-    const char* script;
-    const char* direction;
-    const char* language;
+    FONSshapingRes* result;
+    void* script;
+    void* direction;
+    void* language;
     int it;
 };
```

Comment:
For shaping I would expose the harfbuzz types here. The user needs to have harfbuzz anyway. Just hide fonsSetShaping within FONS_USE_HARFBUZZ. Unless you plan to add more shaping engines. 
hb_script, hb_lang and direction are simple value types iirc - so they should better be kept as value to avoid pointer chasing.